src/python/pants/engine/target.py

```diff
@@ -2719,15 +2722,16 @@ class DependenciesRuleActionRequest:
 
 
 @dataclass(frozen=True)
-class DependenciesRuleAction:
-    """Maps all dependencies to their respective dependency rule action of a origin target address.
+class DependenciesRuleApplication:
+    """Maps all dependencies to their respective dependency rule application of a origin target
+    address.
```

Comment:
@cognifloyd What do you make of this rename?
It's still has the singular vs plural issue tho..